scripts/macsecshow

```diff
@@ -0,0 +1,231 @@
+
+import os
+import swsssdk
+import subprocess
+import sys
+
+import click
+from collections import namedtuple
+from utilities_common.netstat import STATUS_NA
+
+MACSEC_PORT_TABLE_PREFIX = "MACSEC_PORT_TABLE:"
+
+COUNTER_TABLE_PREFIX = "COUNTERS:"
+COUNTERS_MACSEC_FLOW_TX_NAME_MAP = "COUNTERS_MACSEC_FLOW_TX_NAME_MAP"
+COUNTERS_MACSEC_FLOW_RX_NAME_MAP = "COUNTERS_MACSEC_FLOW_RX_NAME_MAP"
+COUNTERS_MACSEC_SA_TX_NAME_MAP   = "COUNTERS_MACSEC_SA_TX_NAME_MAP"
+COUNTERS_MACSEC_SA_RX_NAME_MAP   = "COUNTERS_MACSEC_SA_RX_NAME_MAP"
+
+FlowStats = namedtuple("FlowStats", "ctl, untagged, tx_too_long, rx_no_tag,\
+                       rx_bad_tag, rx_no_sci, rx_unk_sci, rx_overrun,")
+
+flow_stat_names = (
+    'SAI_MACSEC_FLOW_STAT_CONTROL_PKTS',
+    'SAI_MACSEC_FLOW_STAT_PKTS_UNTAGGED',
+    'SAI_MACSEC_FLOW_STAT_OUT_PKTS_TOO_LONG',
+    'SAI_MACSEC_FLOW_STAT_IN_PKTS_NO_TAG',
+    'SAI_MACSEC_FLOW_STAT_IN_PKTS_BAD_TAG',
+    'SAI_MACSEC_FLOW_STAT_IN_PKTS_NO_SCI',
+    'SAI_MACSEC_FLOW_STAT_IN_PKTS_UNKNOWN_SCI',
+    'SAI_MACSEC_FLOW_STAT_IN_PKTS_OVERRUN'
+)
+
+SaStats = namedtuple("SaStats", "b_enc, b_prot, tx_p_enc, tx_p_prot, rx_p_ok,\
+                     rx_p_unchecked, rx_p_delayed, rx_p_late, rx_p_invalid,\
+                     rx_p_not_valid, rx_p_not_using_sa, rx_p_unused_sa")
+
+sa_stat_names = (
+    'SAI_MACSEC_SA_STAT_OCTETS_ENCRYPTED',
+    'SAI_MACSEC_SA_STAT_OCTETS_PROTECTED',
+    'SAI_MACSEC_SA_STAT_OUT_PKTS_ENCRYPTED',
+    'SAI_MACSEC_SA_STAT_OUT_PKTS_PROTECTED',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_OK',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_UNCHECKED',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_DELAYED',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_LATE',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_INVALID',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_NOT_VALID',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_NOT_USING_SA',
+    'SAI_MACSEC_SA_STAT_IN_PKTS_UNUSED_SA'
+)
+
+def get_macsec_port_list(appl_db, intf_name):
+    """
+       Get the macsec port names from database.
+    """
+    if intf_name is None:
+        port_keys = appl_db.keys(appl_db.APPL_DB, MACSEC_PORT_TABLE_PREFIX+"*")
+    else:
+        port_keys = appl_db.keys(appl_db.APPL_DB, MACSEC_PORT_TABLE_PREFIX+"%s" % intf_name)
+    if port_keys is None:
+        return None
+    macsec_ports = [key[len(MACSEC_PORT_TABLE_PREFIX):] for key in port_keys]
+    return macsec_ports
+
+class MacsecConn(object):
+    def display_connections(self, intf_name):
+        """
+            Get MACsec info from wpa_cli.
+        """
+        macsec_cmd = 'docker exec -it macsec wpa_cli -g /var/run/' + intf_name + ' IFNAME=' + intf_name + ' macsec'
+        click.echo("Interface: {} ".format(intf_name))
+        p = subprocess.Popen(macsec_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+        (output, err) = p.communicate()
+        rc = p.wait()
+        if rc == 0:
+            click.echo(output)
+        else:
+            click.echo("Error retrieving MACsec connections: {} ".format(err))
```

Comment:
I think it's better to fetch the connections from APP DB instead of wpa_cli. Because in the future, maybe we will use multiple wpa_supplicant processes to one port for Primary/Fallback key feature. Or we can use the static SAK directly in APP DB for debugging.
So, I think it's better to believe the data in APP DB for more flexibility.

Reply:
This PR is here for contribution to the community in its current state.  Only bug fixes will be made to this PR at this point.  Any major functionality changes will not be made in the context of this PR.  If a different implementation of the "show macsec connections" command is preferred, feel free to use the changes in this PR as a basis for another PR and I will close this one.